proxy/http2/Http2ConnectionState.h

```diff
@@ -38,7 +38,7 @@ enum Http2SendADataFrameResult {
   HTTP2_SEND_A_DATA_FRAME_DONE       = 3,
 };
 
-enum Http2ShutdownState { NOT_INITIATED, INITIATED, IN_PROGRESS };
+enum Http2ShutdownState { NOT_PLANNED, NOT_INITIATED, INITIATED, IN_PROGRESS };
```

Comment:
I don't get it. Are you suggesting any of these?
- Keep `Http2ConnectionState` alive until the event processed
- Make it cancelable and cancel the event when `Http2ConnectoinState` dies

Is it guaranteed that `Http2ConnectionState` is still alive when  `HTTP2_SESSION_EVENT_RECV` is going to be processed?